### PR TITLE
Updated docs for clarity

### PR DESCRIPTION
**Pull request by Linardos**
## Proposed Changes

- in_memory explanation added
- cleaned up setup.md a bit 

## Checklist

- [ ] [`CONTRIBUTING`](https://github.com/mlcommons/GaNDLF/blob/master/CONTRIBUTING.md) guide has been followed.
- [ ] PR is based on the [current GaNDLF master ](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/keeping-your-local-repository-in-sync-with-github/syncing-your-branch-in-github-desktop?platform=windows).
- [ ] Non-breaking change (does **not** break existing functionality): provide **as many** details as possible for _any_ breaking change.
- [ ] Function/class source code documentation added/updated (ensure `typing` is used to provide type hints, including and not limited to using `Optional` if a variable has a pre-defined value).
- [ ] Code has been [blacked](https://github.com/psf/black#usage) for style consistency and linting.
- [ ] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/mlcommons/GaNDLF/blob/master/GANDLF/version.py).
- [ ] If adding a git submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/mlcommons/GaNDLF/blob/master/pyproject.toml) file.
- [ ] [Usage documentation](https://github.com/mlcommons/GaNDLF/blob/master/docs) has been updated, if appropriate.
- [ ] Tests added or modified to [cover the changes](https://app.codecov.io/gh/mlcommons/GaNDLF); if coverage is reduced, please give explanation.
- [ ] If customized dependency installation is required (i.e., a separate `pip install` step is needed for PR to be functional), please ensure it is reflected in all the files that control the CI, namely: [python-test.yml](https://github.com/mlcommons/GaNDLF/blob/master/.github/workflows/python-test.yml), and all docker files [[1](https://github.com/mlcommons/GaNDLF/blob/master/Dockerfile-CPU),[2](https://github.com/mlcommons/GaNDLF/blob/devcontainer_build_fix/Dockerfile-CUDA11.6),[3](https://github.com/mlcommons/GaNDLF/blob/master/Dockerfile-ROCm)].
- [ ] The `logging` library is being used and no `print` statements are left.

### CODE REVIEW

File: GANDLF/metrics/classification.py
Comment (Linardos) [inline GANDLF/metrics/classification.py:1]: I dont know why it recognized this as a change, nothing was changed in this script.

File: docs/customize.md [hunk 1]
```diff
@@ -119,17 +119,17 @@ This file contains mid-level information regarding various parameters that can b
...
 - `verbose`: generate verbose messages on console; generally used for debugging.
-- `batch_size`: defines the batch size to be used for training.
-- `in_memory`: this is to enable or disable lazy loading - setting to true reads all data once during data loading, resulting in improvements.
-- `num_epochs`: defines the number of epochs to train for.
-- `patience`: defines the number of epochs to wait for improvement before early stopping.
-- `learning_rate`: defines the learning rate to be used for training.
-- `scheduler`: defines the learning rate scheduler to be used for training, more details are [here](https://github.com/mlcommons/GaNDLF/blob/master/GANDLF/schedulers/__init__.py); can take the following sub-parameters:
+- `batch_size`: batch size to be used for training.
+- `in_memory`: this is to enable or disable lazy loading. If set to True, all data is loaded on RAM once during data loading, resulting in faster training. If set to False, data gets read into RAM on-the-go when needed, which slows down training but lessens the memory load. The latter is recommended if the user's RAM has limited capacity.
```
Comment (sarthakpati) [inline docs/customize.md:123]:
```suggestion
- `in_memory`: this is to enable or disable lazy loading. If set to `True`, all data is loaded onto the RAM at once during the construction of the dataloader (either training/validation/testing), thus resulting in faster training. If set to `False`, data gets read into RAM on-the-go when needed (also called ["lazy loading"](https://en.wikipedia.org/wiki/Lazy_loading)), which slows down training but lessens the memory load. The latter is recommended if the user's RAM has limited capacity.
```

File: docs/customize.md [hunk 2]
```diff
@@ -119,17 +119,17 @@ This file contains mid-level information regarding various parameters that can b
...
+- `num_epochs`: number of epochs to train for.
+- `patience`: number of epochs to wait for improvement before early stopping.
```
Comment (sarthakpati) [inline docs/customize.md:125]:
```suggestion
- `patience`: number of epochs to wait for improvement in the validation loss before early stopping.
```

File: docs/setup.md [hunk 3]
```diff
@@ -22,64 +22,81 @@ Alternatively, you can run GaNDLF via [Docker](https://www.docker.com/). This ne
...
-# (venv_gandlf) $> pip install torch==2.3.1 torchvision==0.18.1 torchaudio==2.3.1 --index-url https://download.pytorch.org/whl/cu121
-## CUDA 11.8
-# (venv_gandlf) $> pip install torch==2.3.1 torchvision==0.18.1 torchaudio==2.3.1 --index-url https://download.pytorch.org/whl/cu118
-## ROCm 6.0
-# (venv_gandlf) $> pip install torch==2.3.1 torchvision==0.18.1 torchaudio==2.3.1 --index-url https://download.pytorch.org/whl/rocm6.0
-## CPU-only
-# (venv_gandlf) $> pip install torch==2.3.1 torchvision==0.18.1 torchaudio==2.3.1 --index-url https://download.pytorch.org/whl/cpu
+```
+
+You may install pytorch to be compatible with CUDA, ROCm, or CPU-only. An exhaustive list of PyTorch installations can be found here: https://pytorch.org/get-started/previous-versions/#v210
```
Comment (sarthakpati) [inline docs/setup.md:35]:
```suggestion
You may install pytorch to be compatible with CUDA, ROCm, or CPU-only. An exhaustive list of PyTorch installations for the specific version compatible with GaNDLF can be found here: https://pytorch.org/get-started/previous-versions/#v231
```

File: docs/setup.md [hunk 4]
```diff
@@ -22,64 +22,81 @@ Alternatively, you can run GaNDLF via [Docker](https://www.docker.com/). This ne
...
+```bash
+ pip install torch==2.2.1 torchvision==0.17.1 torchaudio==2.2.1 --index-url https://download.pytorch.org/whl/cu118
+```
+- ROCm 5.7
+```bash
+ pip install torch==2.2.1 torchvision==0.17.1 torchaudio==2.2.1 --index-url https://download.pytorch.org/whl/rocm5.7
+```
+- CPU-only
+```bash
+ pip install torch==2.2.1 torchvision==0.17.1 torchaudio==2.2.1 --index-url https://download.pytorch.org/whl/cpu
```
Comment (sarthakpati) [inline docs/setup.md:51]:
```suggestion
- CUDA 12.1
```bash
(venv_gandlf) $>pip install torch==2.3.1 torchvision==0.18.1 torchaudio==2.3.1 --index-url https://download.pytorch.org/whl/cu121
```
- CUDA 11.8
```bash
(venv_gandlf) $>pip install torch==2.3.1 torchvision==0.18.1 torchaudio==2.3.1 --index-url https://download.pytorch.org/whl/cu118
```
- ROCm 6.0
```bash
(venv_gandlf) $>pip install torch==2.3.1 torchvision==0.18.1 torchaudio==2.3.1 --index-url https://download.pytorch.org/whl/rocm6.0
```
- CPU-only
```bash
(venv_gandlf) $>pip install torch==2.3.1 torchvision==0.18.1 torchaudio==2.3.1 --index-url https://download.pytorch.org/whl/cpu
```

File: docs/setup.md [hunk 5]
```diff
@@ -22,64 +22,81 @@ Alternatively, you can run GaNDLF via [Docker](https://www.docker.com/). This ne
...
-(venv_gandlf) $> pip install -e .
+git clone https://github.com/mlcommons/GaNDLF.git
+cd GaNDLF
+pip install -e .
 ```
 
+Test your installation:
+```bash
+gandlf --version
+```
```
Comment (sarthakpati) [inline docs/setup.md:99]: I think the `(venv_gandlf) $>` should be there, which explicitly shows that this needs to go in the same virtual environment as the one the user has created. This removed any doubts.

**Review by sarthakpati — CHANGES_REQUESTED** [review]

File: docs/setup.md (hunk 5 above)
Comment (sarthakpati) [reply docs/setup.md:87]: Hey @Linardos, I didn't see this getting added but the comment was resolved - was that intentional?

Comment (Linardos) [reply docs/setup.md:87]: sorry, I thought it was taken care of with one of the updates you made that I accepted. I'll add it in

Comment (sarthakpati) [reply docs/setup.md]: Thanks!

File: docs/setup.md [hunk 6]
```diff
@@ -22,64 +22,69 @@ Alternatively, you can run GaNDLF via [Docker](https://www.docker.com/). This ne
...
 ### Install from Package Managers
 
 This option is recommended for most users, and allows for the quickest way to get started with GaNDLF.
 
 ```bash
-# continue from previous shell
-(venv_gandlf) $> pip install gandlf # this will give you the latest stable release
-## you can also use conda
-# (venv_gandlf) $> conda install -c conda-forge gandlf -y
+(venv_gandlf) $>pip install gandlf # this will give you the latest stable release
```
Comment (sarthakpati) [inline docs/setup.md:56]:
```suggestion
(venv_gandlf) $> pip install gandlf # this will give you the latest stable release
```

File: docs/setup.md [hunk 7]
```diff
@@ -22,64 +22,69 @@ Alternatively, you can run GaNDLF via [Docker](https://www.docker.com/). This ne
...
+```
+You can also use conda
+```bash
+(venv_gandlf) $>conda install -c conda-forge gandlf -y
```
Comment (sarthakpati) [inline docs/setup.md:60]:
```suggestion
(venv_gandlf) $> conda install -c conda-forge gandlf -y
```

File: docs/setup.md [hunk 8]
```diff
@@ -22,64 +22,69 @@ Alternatively, you can run GaNDLF via [Docker](https://www.docker.com/). This ne
...
 ```
 
 If you are interested in running the latest version of GaNDLF, you can install the nightly build by running the following command:
 
 ```bash
-# continue from previous shell
-(venv_gandlf) $> pip install --pre gandlf
-## you can also use conda
-# (venv_gandlf) $> conda install -c conda-forge/label/gandlf_dev -c conda-forge gandlf -y
+(venv_gandlf) $>pip install --pre gandlf
```
Comment (sarthakpati) [inline docs/setup.md:66]:
```suggestion
(venv_gandlf) $> pip install --pre gandlf
```

File: docs/setup.md [hunk 9]
```diff
@@ -22,64 +22,69 @@ Alternatively, you can run GaNDLF via [Docker](https://www.docker.com/). This ne
...
 ```
 
+You can also use conda
+```bash
+(venv_gandlf) $>conda install -c conda-forge/label/gandlf_dev -c conda-forge gandlf -y
```
Comment (sarthakpati) [inline docs/setup.md:71]:
```suggestion
(venv_gandlf) $> conda install -c conda-forge/label/gandlf_dev -c conda-forge gandlf -y
```

File: docs/setup.md [hunk 10]
```diff
@@ -22,64 +22,69 @@ Alternatively, you can run GaNDLF via [Docker](https://www.docker.com/). This ne
...
 Use this option if you want to [contribute to GaNDLF](https://github.com/mlcommons/GaNDLF/blob/master/CONTRIBUTING.md), or are interested to make other code-level changes for your own use.
 
 ```bash
-# continue from previous shell
-(venv_gandlf) $> git clone https://github.com/mlcommons/GaNDLF.git
-(venv_gandlf) $> cd GaNDLF
-(venv_gandlf) $> pip install -e .
+(venv_gandlf) $>git clone https://github.com/mlcommons/GaNDLF.git
+(venv_gandlf) $>cd GaNDLF
+(venv_gandlf) $>pip install -e .
```
Comment (sarthakpati) [inline docs/setup.md:81]:
```suggestion
(venv_gandlf) $> git clone https://github.com/mlcommons/GaNDLF.git
(venv_gandlf) $> cd GaNDLF
(venv_gandlf) $> pip install -e .
```

File: docs/setup.md [hunk 11]
```diff
@@ -22,64 +22,69 @@ Alternatively, you can run GaNDLF via [Docker](https://www.docker.com/). This ne
...
 ```
 
+Test your installation:
+```bash
+(venv_gandlf) $>gandlf --version
```
Comment (sarthakpati) [inline docs/setup.md:86]:
```suggestion
(venv_gandlf) $> gandlf verify-install
```
